src/client/dfs/dfs.c

```diff
@@ -1688,6 +1688,65 @@ dfs_open(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
 	return rc;
 }
 
+int
+dfs_dup(dfs_t *dfs, dfs_obj_t *obj, dfs_obj_t **_new_obj)
+{
+	dfs_obj_t	*new_obj;
+	int		rc;
+
+	if (dfs == NULL || !dfs->mounted)
+		return -DER_INVAL;
+	if (obj == NULL || _new_obj == NULL)
+		return -DER_INVAL;
+
+	D_ALLOC_PTR(new_obj);
+	if (new_obj == NULL)
+		return -DER_NOMEM;
+
+	switch (obj->mode & S_IFMT) {
+	case S_IFDIR:
+		rc = daos_obj_open(dfs->coh, obj->oid, DAOS_OO_RW,
+				   &new_obj->oh, NULL);
+		if (rc)
+			D_GOTO(err, rc);
+		break;
+	case S_IFREG:
+	{
+		char		buf[1024];
+		daos_iov_t	ghdl;
+
+		daos_iov_set(&ghdl, buf, 1024);
```

Comment:
I thought we'd recently added a check to crt_iov_set() that disallowed IOVs with the values on the stack, does doas_iov_set() not have that check?

Reply:
nope.
static inline void
daos_iov_set(daos_iov_t *iov, void *buf, daos_size_t size)                                                                                                                  
{
        iov->iov_buf = buf;
        iov->iov_len = iov->iov_buf_len = size;
}